src/main/kotlin/com/terraformation/backend/tracking/ObservationService.kt

```diff
@@ -437,6 +439,14 @@ class ObservationService(
       throw IllegalArgumentException("Observed time is in the future")
     }
 
+    if (observationType == ObservationType.BiomassMeasurements && plants.isNotEmpty()) {
+      throw IllegalArgumentException("Biomass observations must not contain plants")
+    }
+
+    if (observationType == ObservationType.BiomassMeasurements && biomassDetails == null) {
+      throw IllegalArgumentException("Biomass observations must contain biomass details")
+    }
```

Comment:
Might make sense to validate that `biomassDetails` is null for monitoring observations. Maybe something more like
```kotlin
when (observationType) {
  ObservationType.BiomassMeasurements -> {
    if (plants.isNotEmpty()) { throw ... }
    if (biomassDetails == null) { throw ... }
  }
  ObservationType.Monitoring -> {
    if (biomassDetails != null) { throw ... }
  }
}
```